### PR TITLE
HPC: Use repo name with build number from 'trigger_hpc'

### PR DESCRIPTION
**Pull request by Soulofdestiny**

### CODE REVIEW

**Comment by coolo** [comment]
which makes testing pretty much invalid. Because restarting the test results of build 10 will not test build 10.

File: tests/hpc/install.pm [hunk 1]
```diff
@@ -30,9 +30,8 @@ sub run() {
     script_run 'systemctl stop packagekit.service';
     # hpc channels
     my $arch     = get_var('ARCH');
-    my $build    = get_var('BUILD_HPC');
     my $reponame = "SLE-Module-HPC12";
-    assert_script_run "zypper ar -f http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/GA/images/repo/SLE-12-Module-HPC-POOL-$arch-Build$build-Media1/ $reponame";
+    assert_script_run "zypper ar -f http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/GA/images/repo/SLE-12-Module-HPC-POOL-$arch-Media1/ $reponame";
```
Comment (sysrich) [inline tests/hpc/install.pm]: Please ensure the test runs in a way that re-running the test will test the same build as expected

Comment (Soulofdestiny) [inline tests/hpc/install.pm]: Right, I didn't think about that...

Comment (okurz) [inline tests/hpc/install.pm]: so close?

**Comment by Soulofdestiny** [comment]
It's now triggered as $build_hpc@$build_sle with every new hpc build
https://openqa.suse.de/tests/618487

so this commit should work and make the test pass

**Comment by coolo** [comment]
What has the triggering to do with what repo you use? if you schedule FOO@BAR; it will still use some random module. And if you retrigger OLD@GRANDPA, it will use the same random module - and *not* OLD@GRANDPA. You're *not* testing the repo you think you do.

And unfortunately we have no way to set the BUILD *actually* tested from within the job.

**Comment by okurz** [comment]
So, as there is only one directory now for HPC on dist.suse.de which is not named with a build number in the directory name I assume we can do the following

* detect that the directory changed in the jenkins job http://lord.arch:8080/job/trigger_HPC_on_osd/ (DONE)
* sync the directory to o.s.d to be available over ftp://o.s.d (we can do that in the same jenkins job)
* register the asset within openQA to ensure old copies will be deleted (How to do that?)
* use either the ftp://o.s.d link or openQA repo asset within the test to test exactly the build we triggered

Does this make sense?
As an alternative I guess we could extend `rsync.pl` but as the shared feeling is that `rsync.pl` is becoming too complicated I figured HPC would be the good example to try something else.

**Comment by sysrich** [comment]
That makes sense, and that is what you should be doing.. it's essential to ensure that our tests run against the builds they think they are running against.

**Comment by Soulofdestiny** [comment]
@okurz

**Comment by Soulofdestiny** [comment]
LGTM by okurz

**Pull request merged**